test/test.cpp

```diff
@@ -213,40 +216,44 @@ TEST_CASE("Smooshable", "[smooshable]") {
   0.3,  0.37,
   0.29, 0.41,
   0.11, 0.97;
+  double scaler_a = 50;
+  double scaler_b = 50;
   Eigen::MatrixXd correct_AB_marginal(2,2);
   correct_AB_marginal <<
-  0.50*0.3+0.71*0.29+0.13*0.11, 0.50*0.37+0.71*0.41+0.13*0.97,
-  0.29*0.3+0.31*0.29+0.37*0.11, 0.29*0.37+0.31*0.41+0.37*0.97;
+  (0.50*0.3+0.71*0.29+0.13*0.11)*50*50, (0.50*0.37+0.71*0.41+0.13*0.97)*50*50,
+  (0.29*0.3+0.31*0.29+0.37*0.11)*50*50, (0.29*0.37+0.31*0.41+0.37*0.97)*50*50;
   Eigen::MatrixXd correct_AB_viterbi(2,2);
   correct_AB_viterbi <<
-  0.71*0.29, 0.71*0.41,
-  0.31*0.29, 0.37*0.97;
+  0.71*0.29*50*50, 0.71*0.41*50*50,
+  0.31*0.29*50*50, 0.37*0.97*50*50;
   Eigen::MatrixXi correct_AB_viterbi_idx(2,2);
   correct_AB_viterbi_idx <<
   1,1,
   1,2;
 
-  Smooshable s_A = Smooshable(A);
-  Smooshable s_B = Smooshable(B);
+  Smooshable s_A = Smooshable(A, scaler_a);
+  Smooshable s_B = Smooshable(B, scaler_b);
   Smooshable s_AB;
   Eigen::MatrixXi AB_viterbi_idx;
   std::tie(s_AB, AB_viterbi_idx) = Smoosh(s_A, s_B);
 
-  REQUIRE(s_AB.marginal() == correct_AB_marginal);
-  REQUIRE(s_AB.viterbi() == correct_AB_viterbi);
+  REQUIRE(s_AB.marginal().isApprox(correct_AB_marginal));
+  REQUIRE(s_AB.viterbi().isApprox(correct_AB_viterbi));
   REQUIRE(AB_viterbi_idx == correct_AB_viterbi_idx);
+  REQUIRE(s_AB.scaler() == 50*50);
 
   Eigen::MatrixXd C(2,1);
   C <<
   0.89,
   0.43;
-  Smooshable s_C = Smooshable(C);
+  double scaler_c = 50;
```

Comment:
Sure thing, I think I tried different scalers and just copy/pasted my "double scaler = 50" statements around the code afterwards. I can change that.